cdap-app-fabric/src/main/java/io/cdap/cdap/internal/metadata/MetadataConsumerSubscriberService.java

```diff
@@ -248,11 +247,18 @@ public void processMessage(MetadataMessage message, StructuredTableContext conte
                  message, t);
         return;
       }
-      // create ProgramRun and LineageInfo for MetadataConsumer
-      long startTimeMs = RunIds.getTime(programRunId.getRun(), TimeUnit.MILLISECONDS);
-      long endTimeMs = System.currentTimeMillis();
-      ProgramRun run = getProgramRunForConsumer(programRunId, startTimeMs, endTimeMs);
-      LineageInfo info = getLineageInfoForConsumer(fieldLineageInfo, startTimeMs, endTimeMs);
+      ProgramRun run;
+      LineageInfo info;
+      try {
+        // create ProgramRun and LineageInfo for MetadataConsumer
+        long startTimeMs = RunIds.getTime(programRunId.getRun(), TimeUnit.MILLISECONDS);
+        long endTimeMs = System.currentTimeMillis();
+        run = getProgramRunForConsumer(programRunId, startTimeMs, endTimeMs);
+        info = getLineageInfoForConsumer(fieldLineageInfo, startTimeMs, endTimeMs);
+      } catch (Throwable t) {
+        LOG.warn("Error while processing field-lineage information received from TMS. Ignoring : {}", message, t);
+        return;
```

Comment:
do we have a unit test?

Reply:
The right way to unit test a public class with private static inner classes is to call the public class and verify the effects. Since the effect here is calling a consumer module that is not part of cdap platform, unit testing becomes tricky.